### PR TITLE
os.scandir refinement

### PR DESCRIPTION
**Pull request by mpajkowski**
Added several changes to achieve better compatibility with CPython [[doc](https://docs.python.org/3/library/os.html#os.scandir)]:

* Make ```path``` an optional arg, set ```"."``` default argument,
* Implement missing methods:
    * context manager - ```__enter__```, ```__exit__```
    * ```close```.

### CODE REVIEW

File: vm/src/stdlib/os.rs [hunk 1]
```diff
@@ -528,6 +528,7 @@ impl DirEntryRef {
 #[derive(Debug)]
 struct ScandirIterator {
     entries: RefCell<fs::ReadDir>,
+    exhausted: RefCell<bool>,
```
Comment (coolreader18) [inline vm/src/stdlib/os.rs]: This could just be a `Cell<bool>`

**Comment by mpajkowski** [comment]
Snippet tests on Windows fails because of  missing ```nt``` module in shutil.

Should I remove ```shutil``` import from this test module or remove/comment out ```shutil.disk_usage``` function and create an issue for missing ```nt```?

**Comment by palaviv** [comment]
I would recommend removing the `shutil` from this PR. We can fix `shutil` in a separate PR and then change the test to use it.

File: tests/snippets/stdlib_os.py [hunk 2]
```diff
@@ -63,27 +63,37 @@
...
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        for root, dirs, files in os.walk(self.name, topdown=False):
+            for name in files:
+                os.remove(os.path.join(root, name))
+            for name in dirs:
+                to_remove = os.path.join(root, name)
+                if os.path.islink(to_remove):
+                    os.unlink(to_remove)
+                else:
+                    os.rmdir(to_remove)
```
Comment (palaviv) [inline tests/snippets/stdlib_os.py]: This will fail if the directory contains files in it. I think keeping the temporary folder for now is OK.

Comment (mpajkowski) [reply tests/snippets/stdlib_os.py]: The problem with keeping this directory is that the timestamp does not have enough time to get incremented during executing that tests - and simply tests fail.

I'll remove this ```__exit__``` code and try some manipulating with dirname.

**Comment by windelbouwman** [comment]
@palaviv do you think this work is in good shape to merge?

**Comment by palaviv** [comment]
Looks good

**Pull request merged**